pkg/action/discover.go

```diff
@@ -40,9 +40,12 @@ var actionsSubdir = strings.Join([]string{"", actionsDirname, ""}, string(filepa
 
 func (ad *yamlDiscovery) isValid(path string, d fs.DirEntry) bool {
 	i := strings.LastIndex(path, actionsSubdir)
-	return !d.IsDir() &&
-		i != -1 &&
-		strings.Count(path[i+len(actionsSubdir):], string(filepath.Separator)) == 1 && // Nested actions are not allowed.
+
+	if d.IsDir() || i == -1 || isHiddenFile(path) {
```

Comment:
It's fine to check it here, but we need to improve it a bit. 
In the function `findFiles`, inside `fs.WalkDir`, we must return if the path is a hidden directory to not traverse the hidden tree for better discovery performance.

We need a condition like
```go
err := fs.WalkDir(ad.fs, ".", func(path string, d fs.DirEntry, err error) error {
...
  if d.IsDir() && isHidden() {
    return fs.SkipDir
  }
```

With such condition, it will never occur in the file check.